xstream/src/java/com/thoughtworks/xstream/converters/extended/AtomicBooleanConverter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2022 XStream Committers.
+ * All rights reserved.
+ *
+ * The software in this package is published under the terms of the BSD
+ * style license a copy of which has been included with this distribution in
+ * the LICENSE.txt file.
+ *
+ * Created on 9 September 2022 by Basil Crow.
+ */
+package com.thoughtworks.xstream.converters.extended;
+
+import com.thoughtworks.xstream.converters.Converter;
+import com.thoughtworks.xstream.converters.MarshallingContext;
+import com.thoughtworks.xstream.converters.SingleValueConverterWrapper;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import com.thoughtworks.xstream.converters.basic.BooleanConverter;
+import com.thoughtworks.xstream.io.HierarchicalStreamReader;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/** Converts an {@link AtomicBoolean}. */
```

Comment:
From an XML PoV an AtomicBoolean is not different to a normal Boolean, therefore the AtomicBooleanConverter should use a similar implementation (and represention of the value).

Reply:
The problem with that approach is that it would break compatibility with existing serialized data that was serialized with the reflection-based converter.